core/src/Lender.sol

```diff
@@ -210,6 +210,23 @@ contract Lender is Ledger {
         emit Repay(msg.sender, beneficiary, amount, units);
     }
 
+    /// @dev Reentrancy guard is critical here! Without it, one could use a flash loan to repay a normal loan.
+    function flash(uint256 amount, IFlashBorrower to, bytes calldata data) external {
+        // Guard against reentrancy
+        uint32 lastAccrualTime_ = lastAccrualTime;
+        require(lastAccrualTime_ != 0, "Aloe: locked");
+        lastAccrualTime = 0;
```

Comment:
This looks good. Ran through this chain of events in my head:
1. Call `flash`
2. `lastAccuralTime_` is set
3. Passes the require statement
4. `lastAccuralTime` is set to 0
5. `onFlashLoan` in the `IFlashBorrower` is triggered
6. Within `onFlashLoan`, we try to call `flash` again 
7. When we re-enter `flash`, `lastAccrualTime_ = 0` 
8. We fail the `require` check

Seems to me like this is will defend us from a re-entrancy attack.

Reply:
Yep! Note on (6), one of the other functions could be called too -- like `deposit` or `repay`. We defend against reentrancy there in the same way: the `_load` function checks that `lastAccrualTime != 0`